src/CatalogService/src/Domain/Entities/Product.cs

```diff
@@ -0,0 +1,43 @@
+﻿namespace CatalogService.Domain.Entities;
+public class Product : BaseEntity
+{
+    public string Name { get; set; }
+
+    public string Description { get; set; }
+
+    public int Image { get; set; }
+
+    public int CatalogId { get; set; }
+
+    public Catalog Catalog { get; set; }
+
+    public decimal Price { get; set; }
+
+    public uint Amount { get; set; }
```

Comment:
As long as you decide to follow DDD, then, let's follow it fully. In the current situation, you have separate actions for AddAmount and RemoveAmount, but nothing prevents us from manually updating this value. Consequently, the Amount setter should be private and populated through the constructor.